src/cim/components/ClusterDeployment/ClusterDeploymentHostsSelectionBasic.tsx

```diff
@@ -1,124 +1,74 @@
 
 type ClusterDeploymentHostsSelectionBasicProps = {
   availableAgents: AgentK8sResource[];
-  isSNOCluster: boolean;
-};
-
-const getHostCountWarningText = (hostsSelected: number) => {
-  switch (hostsSelected) {
-    case 0:
-      return 'No host is selected.';
-    case 1:
-      return 'Only 1 host is selected.';
-  }
-  return `Only ${hostsSelected} hosts are selected.`;
+  isSNOCluster?: boolean;
 };
 
 const ClusterDeploymentHostsSelectionBasic: React.FC<ClusterDeploymentHostsSelectionBasicProps> = ({
   isSNOCluster,
   availableAgents,
 }) => {
-  const { setFieldValue, values } = useFormikContext<ClusterDeploymentHostsSelectionValues>();
-  const { hostCount, locations, autoSelectedHostIds } = values;
-
-  const [matchingAgents, selectedAgents] = React.useMemo(() => {
-    const mAgents = availableAgents.filter((agent) => {
-      const agentLocation =
-        agent.metadata?.labels?.[AGENT_LOCATION_LABEL_KEY] || AGENT_NOLOCATION_VALUE;
-      return locations.length ? locations.includes(agentLocation) : true;
-    });
-    const sAgents = mAgents.filter((a) => autoSelectedHostIds.includes(a.metadata?.uid || ''));
-    return [mAgents, sAgents];
-  }, [availableAgents, locations, autoSelectedHostIds]);
+  const { setFieldValue } = useFormikContext<ClusterDeploymentHostsSelectionValues>();
+  const { matchingAgents, selectedAgents, hostCount } = useAgentsAutoSelection(availableAgents);
 
   React.useEffect(() => {
-    const ids = matchingAgents.map((a) => a.metadata?.uid).splice(0, hostCount);
-    if (!_.isEqual(ids, autoSelectedHostIds)) {
-      setTimeout(() => setFieldValue('autoSelectedHostIds', ids, true));
+    if (hostCount === 3) {
+      setFieldValue('useMastersAsWorkers', true);
     }
-  }, [matchingAgents, setFieldValue, autoSelectedHostIds, hostCount]);
+  }, [hostCount, setFieldValue]);
 
   return (
     <>
-      <GridItem span={12} lg={10} xl={9} xl2={7}>
-        <NumberInputField
-          labelIcon={<HostCountLabelIcon />}
-          idPostfix="hostcount"
-          name="hostCount"
-          isRequired
-          minValue={isSNOCluster ? 1 : HOSTS_MIN_COUNT}
-          maxValue={isSNOCluster ? 1 : HOSTS_MAX_COUNT}
-          isDisabled={isSNOCluster}
-          onChange={(newValue) => {
-            if (newValue === 4) {
-              newValue = values.hostCount >= 4 ? 3 : 5;
-            }
-            setFieldValue('hostCount', newValue);
-            if (newValue === 3) {
-              setFieldValue('useMastersAsWorkers', true);
-            }
-          }}
-        />
-      </GridItem>
-
-      {/* That field is not supported ATM - (requires MGMT-7677) */}
-      <GridItem>
-        <CheckboxField
-          idPostfix="mastersasworkers"
-          name="useMastersAsWorkers"
-          label="Run workloads on control plane (master) hosts"
-          isDisabled={hostCount < 5}
-        />
-      </GridItem>
+      <Grid hasGutter>
+        <GridItem span={12} lg={10} xl={9} xl2={7}>
```

Comment:
same here - needs to be verified in ACM/YAML editor open scenario

Reply:
I've fixed the vertical spacing and removed the largest breakpoint to avoid locations field to be too small in case when yaml editor is open